components/brave_rewards/resources/ui/components/adsBox.tsx

```diff
@@ -24,10 +24,30 @@ import * as rewardsActions from '../actions/rewards_actions'
 // Assets
 const adsDisabledIcon = require('../../img/ads_disabled.svg')
 
+interface State {
```

Comment:
why do we need internal state? we already have this data in rewardsData

Reply:
I did this to avoid the use of something like `firstLoad` for value precision. Having an internal state that is kept in sync with `rewardsData` lessens things like form control latency.

If this is a huge issue I can remove it but I think it's helpful

Reply:
@NejcZdovc state removed